pkg/filter/pipelinerun.go

```diff
@@ -0,0 +1,156 @@
+	// TektonPipelineRunTriggeredBuilds contains references for all Builds triggered, JSON formatted
+	TektonPipelineRunTriggeredBuilds = fmt.Sprintf("%s/pipelinerun-triggered-builds", Prefix)
+)
+
+// searchBuildRunForRunOwner inspect the object owners for Tekton Run and returns it, otherwise nil.
+func searchBuildRunForRunOwner(br *v1alpha1.BuildRun) *types.NamespacedName {
+	for _, ownerRef := range br.OwnerReferences {
+		if ownerRef.APIVersion == constants.TektonAPIv1alpha1 && ownerRef.Kind == "Run" {
+			return &types.NamespacedName{Namespace: br.GetNamespace(), Name: ownerRef.Name}
+		}
+	}
+	return nil
+}
+
+// filterBuildRunOwnedByRun filter out BuildRuns objects not owned by Tekton Run.
+func filterBuildRunOwnedByRun(obj interface{}) bool {
+	br, ok := obj.(*v1alpha1.BuildRun)
+	if !ok {
+		return false
+	}
+	return searchBuildRunForRunOwner(br) != nil
+}
+
+// pipelineRunReferencesShipwright checks if the informed PipelineRun is reffering to a Shipwright
+// resource via TaskRef.
+func pipelineRunReferencesShipwright(pipelineRun *tknv1beta1.PipelineRun) bool {
+	if pipelineRun.Status.PipelineSpec == nil {
+		return false
+	}
+	for _, task := range pipelineRun.Status.PipelineSpec.Tasks {
+		if task.TaskRef == nil {
+			continue
+		}
+		if task.TaskRef.APIVersion == constants.ShipwrightAPIVersion {
+			return true
+		}
+	}
+	return false
+}
+
+// EventFilterPredicate predicate filter for the basic inspections in the object, filtering only what
+// needs to go through reconciliation. PipelineRun objects referencing Custom-Tasks are also skipped.
+func EventFilterPredicate(obj client.Object) bool {
+	logger := logr.New(log.Log.GetSink()).
+		WithValues("namespace", obj.GetNamespace(), "name", obj.GetName())
+
+	pipelineRun, ok := obj.(*tknv1beta1.PipelineRun)
+	if !ok {
+		logger.V(0).Error(nil, "Unable to cast object as Tekton PipelineRun")
+		return false
+	}
+
+	if !pipelineRun.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.V(0).Info("Marked for deletion")
+		return false
+	}
+
+	if pipelineRun.Spec.PipelineRef == nil {
+		logger.V(0).Info("Skipping due nil .Spec.PipelineRef")
+		return false
+	}
+
+	if pipelineRun.Status.PipelineSpec == nil {
+		logger.V(0).Info("Skipping due to nil .Status.PipelineSpec")
+		return false
+	}
+
+	if pipelineRunReferencesShipwright(pipelineRun) {
+		logger.V(0).Info("Skipping due to being part of a Custom-Task")
+		return false
+	}
+	return true
+}
+
+// ParsePipelineRunStatus parte the informed object status to extract its status.
+func ParsePipelineRunStatus(
+	ctx context.Context,
+	now time.Time,
+	pipelineRun *tknv1beta1.PipelineRun,
+) (string, error) {
+	switch {
+	case pipelineRun.IsDone():
+		if pipelineRun.Status.GetCondition(apis.ConditionSucceeded).IsTrue() {
+			return tknv1beta1.PipelineRunReasonSuccessful.String(), nil
+		}
+		return tknv1beta1.PipelineRunReasonFailed.String(), nil
+	case pipelineRun.IsCancelled():
+		return tknv1beta1.PipelineRunReasonCancelled.String(), nil
```

Comment:
The Tekton function is imo weird. It checks the spec value (which I would label IsCancelRequested) but not the status that it really has been canceled successfully. I's prefer if we trigger based on the latter.

Reply:
The current approach relies on the Tekton logic directly, I think we should stick with the logic provided by their package directly.